oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistrationRepositoryUtils.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.client.registration;
+
+import org.springframework.util.Assert;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Utility methods for InMemoryClientRegistrationRepository and InMemoryReactiveClientRegistrationRepository.
+ * 
+ * @author David Perez
+ * @since 5.1
+ * @see InMemoryClientRegistrationRepository
+ * @see InMemoryReactiveClientRegistrationRepository
+ */
+public final class ClientRegistrationRepositoryUtils {
+
+	private ClientRegistrationRepositoryUtils() {
+	}
+
+	static Map<String, ClientRegistration> toUnmodifiableMapByRegistrationId(Stream<ClientRegistration> registrations) {
+		return Collections.unmodifiableMap(registrations
+				.peek(registration -> Assert.notNull(registration, "registrations cannot contain null values"))
+				.collect(Collectors.toMap(ClientRegistration::getRegistrationId, Function.identity())));
+	}
+
+	static <T> T checkEmptyRegistrations(T registrations, BiConsumer<? super T, String> checker){
```

Comment:
Rather than adding a separate method, I think it would be good to validate in the conversion to a Map

Reply:
Thank you very much for the comments, I find them very instructive. If you run the PR unit tests without the main sources, you'll see that the pairs of builders are not fully consistent. Do you think it's better now?